### PR TITLE
Only treat an inlinerecord field as a member of a scope if it's not marked SkipForInlineRecords

### PR DESCRIPTION
**Pull request by lesaltzm**
`ForAll` and other functions can be marked as SkipForInlineRecords. Those functions should not have their scopes used for display names associated with Inline Records. This is a fix for a long-standing bug in PowerApps where we incorrectly rewrite inline records on display name changes, even when they should not be associated with any display names.

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 1]
```diff
@@ -4958,11 +4958,10 @@ private bool IsRecordScopeFieldName(DName name, out Scope scope)
                         DType.TryGetLogicalNameForColumn(scope.Type, name.Value, out maybeLogicalName)))
                     {
                         name = new DName(maybeLogicalName);
-                    }
-
-                    if (scope.Type.TryGetType(name, out var tmpType))
-                    {
-                        return true;
+                        if (scope.Type.TryGetType(name, out var tmpType))
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4961]: >name [](http://example.com/codeflow?start=50&length=4)

new DName(maybeLogicalName)

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 2]
```diff
@@ -4958,11 +4958,10 @@ private bool IsRecordScopeFieldName(DName name, out Scope scope)
                         DType.TryGetLogicalNameForColumn(scope.Type, name.Value, out maybeLogicalName)))
                     {
                         name = new DName(maybeLogicalName);
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4960]: >name = new DName(maybeLogicalName); [](http://example.com/codeflow?start=24&length=35)

You probably don't need this line.

**Comment by MikeStall** [comment]
Can you give a bit more context about this to help educate what it's about?

**Comment by Shpakh-MSFT** [comment]
>                     Contracts.AssertValue(scope);

if(scope.SkipForInlineRecords)
    continue;


---
Refers to: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:4953 in cc3d9a5. [](commit_id = cc3d9a53b5d6fcd039e3224397db1a29e49007a1, deletion_comment = False)

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

**Review by rick-nguyen — APPROVED** [review]

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**